app/models/v2/event.rb

```diff
@@ -3,4 +3,9 @@ class V2::Event < ActiveRecord::Base
 
   validates :description, presence: true
   validates :time_slots, presence: true
+
+  def available_time_slots
+    available_time_slots = time_slots.find_all { |slot| !slot.reservation.present? }
+    available_time_slots || []
+  end
```

Comment:
@ganeshbsub would you please convert this into a named scope? I'm focusing on SMS scheduling today